internal/incident/incident.go

```diff
@@ -119,74 +109,47 @@ func (i *Incident) ProcessEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event
 	return i.notifyContacts(ctx, tx, &ev, causedBy)
 }
 
-func (i *Incident) processSeverityChangedEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event) (types.Int, error) {
-	oldIncidentSeverity := i.Severity()
-	oldSourceSeverity := i.SeverityBySource[ev.SourceId]
-	if oldSourceSeverity == event.SeverityNone {
-		oldSourceSeverity = event.SeverityOK
+func (i *Incident) processSeverityChangedEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event, created bool) (types.Int, error) {
+	oldIncidentSeverity := i.Severity
+	if oldIncidentSeverity == event.SeverityNone {
+		oldIncidentSeverity = event.SeverityOK
 	}
 
-	if oldSourceSeverity == ev.Severity {
+	var causedByHistoryId types.Int
+	newIncidentSeverity := ev.Severity
+	if oldIncidentSeverity == newIncidentSeverity && !created {
 		msg := fmt.Sprintf("Ignoring superfluous %q state event from source %d", ev.Severity.String(), ev.SourceId)
 		i.logger.Warnln(msg)
 
-		return types.Int{}, errors.New(msg)
-	}
-
-	i.logger.Infof(
-		"Source %d severity changed from %s to %s", ev.SourceId, oldSourceSeverity.String(), ev.Severity.String(),
-	)
-
-	history := &HistoryRow{
-		EventID:     utils.ToDBInt(ev.ID),
-		Type:        SourceSeverityChanged,
-		Time:        types.UnixMilli(time.Now()),
-		NewSeverity: ev.Severity,
-		OldSeverity: oldSourceSeverity,
-		Message:     utils.ToDBString(ev.Message),
-	}
-	causedByHistoryId, err := i.AddHistory(ctx, tx, history, true)
-	if err != nil {
-		i.logger.Errorw("Can't insert source severity changed history to the database", zap.Error(err))
-
-		return types.Int{}, errors.New("can't insert source severity changed history to the database")
-	}
-
-	if err = i.AddSourceSeverity(ctx, tx, ev.Severity, ev.SourceId); err != nil {
-		i.logger.Errorw("Failed to upsert source severity", zap.Error(err))
-
-		return types.Int{}, errors.New("failed to upsert source severity")
+		return causedByHistoryId, errors.New(msg)
 	}
 
-	if ev.Severity == event.SeverityOK {
-		delete(i.SeverityBySource, ev.SourceId)
-	}
-
-	newIncidentSeverity := i.Severity()
-	if newIncidentSeverity != oldIncidentSeverity {
-		i.logger.Infof(
-			"Incident severity changed from %s to %s", oldIncidentSeverity.String(), newIncidentSeverity.String(),
-		)
+	if oldIncidentSeverity != newIncidentSeverity {
```

Comment:
Then this should also no longer need this if. So if we get here, shouldn't there always be a change?

Reply:
See the above comment!